pkg/smart/parse.go

```diff
@@ -0,0 +1,329 @@
+package smart
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/pkg/errors"
+)
+
+type ataSmartAttributes struct {
+	Table []struct {
+		ID         int    `json:"id"`
+		Name       string `json:"name"`
+		Value      int    `json:"value"`
+		Worst      int    `json:"worst"`
+		Thresh     int    `json:"thresh"`
+		WhenFailed string `json:"when_failed"`
+		Raw        struct {
+			Value  int    `json:"value"`
+			String string `json:"string"`
+		} `json:"raw"`
+	} `json:"table"`
+}
+
+type nvmeSmartHealthInformationLog struct {
+	CriticalWarning         int   `json:"critical_warning"`
+	Temperature             int   `json:"temperature"`
+	AvailableSpare          int   `json:"available_spare"`
+	AvailableSpareThreshold int   `json:"available_spare_threshold"`
+	PercentageUsed          int   `json:"percentage_used"`
+	DataUnitsRead           int64 `json:"data_units_read"`
+	DataUnitsWritten        int64 `json:"data_units_written"`
+	HostReads               int64 `json:"host_reads"`
+	HostWrites              int64 `json:"host_writes"`
+	ControllerBusyTime      int64 `json:"controller_busy_time"`
+	PowerCycles             int   `json:"power_cycles"`
+	PowerOnHours            int   `json:"power_on_hours"`
+	UnsafeShutdowns         int   `json:"unsafe_shutdowns"`
+	MediaErrors             int   `json:"media_errors"`
+	NumErrLogEntries        int   `json:"num_err_log_entries"`
+}
+
+type parseResult struct {
```

Comment:
Is it the beast approach to define all these as anonymous types?

Reply:
These fields are internal use only to parse `json` output from `smartctl`.